detectable/src/main/java/com/blackduck/integration/detectable/detectables/gradle/inspection/parse/GradleReportTransformer.java

```diff
@@ -24,22 +28,108 @@
 public class GradleReportTransformer {
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
     private final EnumListFilter<GradleConfigurationType> configurationTypeFilter;
+    private final List<CodeLocation> allCodeLocationsWithinRoot = new ArrayList<>();
```

Comment:
Set?

Reply:
Using a Set here would have been ideal (checking to see if a code location for some subProject has aready been created would have been simpler). However, the CodeLocation class did not have a defined method for overriding `equals() `... and without that, inserting into the Set wouldn't have worked as we want (where two code locations are equal if their external IDs are equal). The CodeLocation object is used heavily troughout our code, I was hesitant to make any changes as I would first have to confirm the equals() as I define it would not have unwanted affects elsewhere in the code.